### DISCUSSION TITLE
Ignore `function-bind` npm package in `unstable_allowDynamic` by default for edge

### DISCUSSION THREAD

**Discussion by rattrayalex** [vercel/next.js#51910] [Ideas]
### Goals

1. Allow users to depend on packages like `qs` (or other packages which depend on them, like `stripe`), which contain `function-bind` solely as a polyfill for old environments. 
2. Allow package authors who depend on `qs` (or similar) to claim they support Next.JS and Vercel without caveats, or without needing to tell users to modify their configs. (This is me).

### Non-Goals

You _could_ choose to find other common packages that need to be ignored, and ignore them by default too, but I don't have an opinion there. 

This does not aim to address turning `unstable_allowDynamic` on for user code by default.

### Background

The docs for this feature mention this package: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation so I assume it's quite common. 

`qs` maintainer @ljharb [says he will not drop the `function-bind` dependency](https://github.com/ljharb/qs/issues/476) from `qs` or other libraries he maintains.

### Proposal

Basically make this the default for `edge` users, and if the user specifies their own `unstable_allowDynamic` items, concat them:
```ts
export const config = {
  runtime: 'edge',
  unstable_allowDynamic: [
    '/node_modules/function-bind/**',
  ],
};
```
(though [this may not be sufficient with pnpm](https://github.com/vercel/next.js/issues/51904))

**Comment by ljharb** [comment]
This is 100% fine, fwiw, to replace `require('function-bind')` with `Function.prototype.bind` as long as you support envs with ES5+.

**Comment by RobertCraigie** [comment]
What's not explicitly called out in this discussion so far as well is that, at least as far as I'm aware, the unsafe code in `function-bind` will never actually be executed at runtime as `function-bind` essentially does nothing when `Function.prototype.bind` is already defined, which will always be the case in Edge environments.

**Comment by rattrayalex** [comment]
Resolved in https://github.com/vercel/next.js/pull/52009 🎉
